astropy/coordinates/builtin_frames/icrs_cirs_transforms.py

```diff
@@ -17,12 +17,62 @@
 from ... import _erfa as erfa
 
 from .icrs import ICRS
+from .itrs import ITRS
 from .gcrs import GCRS
 from .cirs import CIRS
 from .hcrs import HCRS
 from .utils import get_jd12
 
 
+"""
+rotational velocity of Earth in radians per (UT1) second
+See Explanatory Supplement to the Astronomical Almanac, ed. P. Kenneth Seidelmann (1992),
+University Science Books.
+"""
+V_EARTH = u.Quantity([0, 0, 7.292115855306589e-5])*u.rad/u.s
+
+
+# helper function to find GCRS position and velocity of observatory
+def get_gcrs_posvel(time, frame):
+    """
+    Given frame with ```obsgeoloc``` and ```obsgeovel```, find GCRS position and velocity.
+
+    ```obsgeoloc``` and ```obsgeovel``` are specified in the ITRS coordinate frame.
+    This function converts the position and velocity into the GCRS coordinate frame.
+
+    Parameters
+    ----------
+    time : `~astropy.time.Time`
+        The time at which to calculate GCRS position and velocity.
+    frame : `~astropy.coordinates.BaseCoordinateFrame
+        The frame to use. Must have ```obsgeoloc``` and ```obsgeovel``` attributes.
+
+    Returns
+    --------
+    pv : np.ndarray
+        GCRS position and velocity in m and m/s
+    """
+    try:
+        obsgeoloc = ITRS(frame.obsgeoloc, obstime=time)
+        obsgeovel = ITRS(frame.obsgeovel, obstime=time)
+    except AttributeError:
+        raise ValueError("frame must have obsgeoloc and obsgeovel attributes set")
+
+    # transform to GCRS. Just a rotation matrix so velocity should be fine?
+    geocentric_frame = GCRS(obstime=time)
+    pos_arr = obsgeoloc.transform_to(geocentric_frame).cartesian.xyz.to(u.m).value
+    vel_arr = obsgeovel.transform_to(geocentric_frame).cartesian.xyz.to(u.m/u.s).value
+
+    # now add rotational velocity of GCRS frame w.r.t to ITRS
+    extra_rotation = np.cross(V_EARTH, np.rollaxis(pos_arr, 0, pos_arr.ndim))
+    vel_arr += np.rollaxis(extra_rotation, -1, 0)
+
+    pv = np.array([pos_arr, vel_arr])
+    if len(pv.shape) > 2:
```

Comment:
If so, maybe explicitly write `if time.ndim > 1:`?
Or add a comment here... (can be commit with [skip ci])